File: go/web/toolkit/button.go
package toolkit

import (
	"github.com/apkumar/go-option"

	"github.com/snapper-labs/immediate/go"
)

type ButtonOptions struct {
	Key string
	OnClick func()
}

func Button(ui *immgo.RenderNode, label string, options ...ButtonOptions) {
	opts := option.Merge(options...)

	desc := immgo.ElementDescription {
		Kind: "itk-button",
		Key: opts.Key,
		Properties: immgo.Properties {
			Attributes: immgo.Attributes {
				"label": label,
			},
			EventHandlers: immgo.EventHandlers {
				"click": func(event interface{}) {
					if opts.OnClick != nil {
						opts.OnClick()
					}
				},
			},
		},
	}


	immgo.Render(ui, desc)
}

File: toolkit/src/stories/Button.ts
import { html } from 'lit-html';
import './button.css';
import '../Button';

export interface ButtonProps {
  label: string;
}

export const Button = ({label}: ButtonProps) => {
  return html`
    <itk-button label="${label}">
    </itk-button>
  `;
};
